analyzers/src/SonarAnalyzer.CSharp/Extensions/CompilationUnitSyntaxExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Extensions
+{
+    public static class CompilationUnitSyntaxExtensions
+    {
+        public static IEnumerable<SyntaxNode> GetTopLevelMainBody(this CompilationUnitSyntax compilationUnit) =>
+            compilationUnit.ChildNodes()
```

Comment:
- I would change this name to `GlobalStatements`. I think it describes better the purpose as my understating is that it gets all the `GlobaStatement` nodes under the `CompilationUnit`.
- I think also `compilationUnit.ChildNodes().Where(x => x.IsKind(SyntaxKind.GlobalStatement))` will also work (but maybe the query in lines 34-35 is actually providing some optimization?). 
- Would it make sense to retrieve the first child of each `GlobalStatement` instead and return that, since my understanding, for example from this [AST](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLACgCWAdsAAR7EC8xGA3KQgtbnvYzrigCzEAyA9gMZgANigCswPLwIAKAJTFcAb1zFVdBrgC+uXACcApv167YxAMpgAtgAch+pTpza2OAlf0RrYfvuIAVDxJlHHRiAEEAqGJFZ00gA==) is that the node of interest is the first child of the `GlobalStatement`.

Reply:
> I would change this name to GlobalStatements. I think it describes better the purpose as my understating is that it gets all the GlobaStatement nodes under the `CompilationUnit`.

I prefer to use `GetTopLevelMainBody` since the intent is to get the Main method body, if later we find some exceptions, or if the syntax tree changes (new features added to the language) we will want to update the method in that regard and we will not focus on `GlobalStatements`.

> I think also compilationUnit.ChildNodes().Where(x => x.IsKind(SyntaxKind.GlobalStatement)) will also work (but maybe the query in lines 34-35 is actually providing some optimization?).

I prefer `TakeWhile(` since the iteration will stop once we finish reading `GlobalStatements`. Currently you cannot have `GlobalStatements` after type declarations start.

> Would it make sense to retrieve the first child of each GlobalStatement instead and return that, since my understanding, for example from this AST is that the node of interest is the first child of the GlobalStatement.

I prefer to let the nodes unchanged for now just to keep things simple. We can see later if they are not needed and if removing them will help in some regard.